apps/web/components/eventtype/EventTeamTab.tsx

```diff
@@ -394,10 +398,29 @@ export const EventTeamTab = ({
     );
   });
   const isManagedEventType = eventType.schedulingType === SchedulingType.MANAGED;
+  const [isOpenAssignmentWarnDialog, setIsOpenAssignmentWarnDialog] = useState<boolean>(false);
+  const [pendingRoute, setpendingRoute] = useState("");
+  const isConfirm = useRef(false);
   const { getValues, setValue } = useFormContext<FormValues>();
   const [assignAllTeamMembers, setAssignAllTeamMembers] = useState<boolean>(
     getValues("assignAllTeamMembers") ?? false
   );
+  const router = useRouter();
+
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
```

Comment:
Is there a reason to have "handleRouteChange" inside "useEffect" ? Why not move it out and wrap it in `useCallback` and then call it on "routeChangeStart"

Reply:
There is no specific reason for it. Do we really need to cache the function since it only runs on route events and checks the assignee numbers?